zulipterminal/ui.py

```diff
@@ -130,15 +125,21 @@ def set_footer_text(
         self.frame.footer.set_text(text)
         self.frame.footer.set_attr_map({None: style})
         self.controller.update_screen()
-        if duration is not None:
-            assert duration > 0
-            time.sleep(duration)
-            self.reset_footer_text()
 
     def reset_footer_text(self) -> None:
         text = self.get_random_help()
         self.set_footer_text(text, "footer")
 
+    def set_footer_text_for_event(self, text: List[Any], style: str = "footer") -> None:
+        self.set_footer_text(text, style)
+
+    def set_footer_text_for_event_duration(
+        self, text: List[Any], duration: float, style: str = "footer"
+    ) -> None:
+        self.set_footer_text_for_event(text, style)
+        time.sleep(duration)
+        self.reset_footer_text()
```

Comment:
This function is not asynch, so blocks for that duration. This breaks existing behavior. For example, try editing an old message that you cannot - the UI just stops updating, before catching up.

We shouldn't make everything asynch, as that puts everything in a thread and makes it a lot more complex, which was one benefit to the existing approach. However, this needs resolving.